Packs/CloudflareWAF/Integrations/CloudflareWAF/CloudflareWAF.py

```diff
@@ -0,0 +1,1212 @@
+        client (Client): Cloudflare API client.
+        args (Dict[str, Any]): Command arguments from XSOAR.
+
+    Returns:
+        CommandResults: outputs, readable outputs and raw response for XSOAR.
+    """
+
+    name = args['name']
+    description = args.get('description')
+
+    response = client.cloudflare_waf_ip_list_create_request(
+        name, description=description)
+
+    output = response['result']
+    readable_output = tableToMarkdown(
+        name='IP list was successfully created.',
+        t=output,
+        headers=['id', 'name', 'description', 'kind', 'num_items', 'num_referencing_filters', 'created_on', 'modified_on'],
+        headerTransform=string_to_table_header
+    )
+    return CommandResults(
+        readable_output=readable_output,
+        outputs_prefix='CloudflareWAF.IpList',
+        outputs_key_field='id',
+        outputs=output,
+        raw_response=response
+    )
+
+
+def cloudflare_waf_ip_list_delete_command(client: Client, args: Dict[str, Any]) -> CommandResults:
+    """ Delete IP-list by the specified ID.
+
+    Args:
+        client (Client): Cloudflare API client.
+        args (Dict[str, Any]): Command arguments from XSOAR.
+
+    Returns:
+        CommandResults: outputs, readable outputs and raw response for XSOAR.
+    """
+
+    list_id = args['id']
+
+    output = client.cloudflare_waf_ip_list_delete_request(list_id)
+
+    return CommandResults(
+        readable_output=f'IP list {list_id} was successfully deleted',
+        raw_response=output
+    )
+
+
+def cloudflare_waf_ip_list_item_create_command(client: Client, args: Dict[str, Any]) -> CommandResults:
+    """ Create a new ip-list items.
+
+    Args:
+        client (Client): Cloudflare API client.
+        args (Dict[str, Any]): Command arguments from XSOAR.
+
+    Returns:
+        CommandResults: outputs, readable outputs and raw response for XSOAR.
+    """
+
+    list_id = args['list_id']
+    items = [{'ip': item} for item in argToList(args.get('items'))]
+
+    response = client.cloudflare_waf_ip_list_item_create_request(list_id, items)
+    output = response['result']
+
+    return CommandResults(
+        readable_output=f'Adding items to the ip-list {list_id} is executing'
+    ), output
+
+
+def cloudflare_waf_ip_list_item_update_command(client: Client, args: Dict[str, Any]) -> CommandResults:
+    """ Replace exist ip-list items with a new items.
+
+    Args:
+        client (Client): Cloudflare API client.
+        args (Dict[str, Any]): Command arguments from XSOAR.
+
+    Returns:
+        CommandResults: outputs, readable outputs and raw response for XSOAR.
+    """
+
+    list_id = args['list_id']
+
+    items = [{'ip': item} for item in argToList(args.get('items'))]
+
+    response = client.cloudflare_waf_ip_list_item_update_request(list_id, items)
+
+    output = response['result']
+
+    return CommandResults(
+        readable_output=f'Replacing items in the IP List {list_id} is executing'
+    ), output
```

Comment:
Is it possible that a few lines here were deleted unintentionally?

Reply:
It is deleted after meeting with you, at your request.